### ISSUE TITLE
debuild -uc -us fails, ubuntu 16.04 (cloned source from github, test passes)

### ISSUE THREAD

**Issue by calocedrus** [Aries85/LightZone#196]
I've successfully built lightzone from source, tested it (ok) but `debuild -uc -us` failed and I have no *lightzone*.deb package anywhere. Trying to find a solution, I've found [this thread](http://lightzoneproject.org/lorum/ubuntu-1304-lightzone-would-not-start) on the lightzone forum and followed what Masahiro (@ktgw0316) recommends but to no avail.

Here are all the steps I took:

```
git clone https://github.com/ktgw0316/LightZone.git
cd LightZone/
sudo apt-get install debhelper devscripts build-essential ant autoconf git-core javahelp2 default-jdk default-jre-headless libjpeg-turbo8-dev libtiff5-dev libx11-dev
export JAVA_HOME=/usr/lib/jvm/java-1.7.0-openjdk-amd64
ant -f linux/build.xml #(build ok)
ant -f linux/build.xml run #(runs ok)
debuild -uc -us # not ok see below
```

[...]

`BUILD SUCCESSFUL Total time: 0 seconds rm -f -rf /home/ludovic/Programs/Photography/LightZone/debian/lightzone/ rm -f configure-stamp build-stamp  dpkg-source -b LightZone dpkg-source: error: can't build with source format '3.0 (quilt)': no upstream tarball found at ../lightzone_4.1.7.orig.tar.{bz2,gz,lzma,xz} dpkg-buildpackage: error: dpkg-source -b LightZone gave error exit status 255 debuild: fatal error at line 1364: dpkg-buildpackage -rfakeroot -d -us -uc failed`

I did execture `debuild -uc -us` from the root (or "master") directory of lightzone. So what is wrong with my installation and/or system?

PS: I couldn't find the issue from github.com/ktgw0316/LightZone which is where one is directed to download the source code, so I've just posted this message on the lightzone forum, after which I've found the link to this issue tracker (on a different github branch from the source). I will update here (github) or there (the forum) as soon as I have a solution or feedback from whichever comes first.

**Comment by ktgw0316** [comment]
You need to download the latest source tarball before creating the package, like this:

```
sudo apt install curl
curl -L https://github.com/Aries85/LightZone/tarball/master > ../lightzone_4.1.7.orig.tar.gz
```

If it works, I'll update the forum comment and Linux build instruction.

**Comment by calocedrus** [comment]
Thank you. This seems to have solved the install problem (with a small detail which may need to be fixed in Ubuntu, see below), but then I get an error when I launch lightzone:
```
$ lightzone 
Starting LightZone version 4.1.7 ...
with options :
java version "1.8.0_60"
Java(TM) SE Runtime Environment (build 1.8.0_60-b27)
Java HotSpot(TM) 64-Bit Server VM (build 25.60-b23, mixed mode)
/usr/bin/lightzone: 65: /usr/bin/lightzone: -XX:MinHeapFreeRatio=30: not found
Error: Could not find or load main class com.lightcrafts.platform.linux.LinuxLauncher
```
Why so? After `ant -f linux/build.xml`, doing `ant -f linux/build.xml run` did launch lightzone without any issue.

In the meantime, here are some detail of install steps, on ubuntu (14.04):
`curl -L https://github.com/Aries85/LightZone/tarball/master > ../lightzone_4.1.7.orig.tar.gz`
didn't work and returned an error:
```
curl: (77) error setting certificate verify locations:
  CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: none
```
To solve this I did
```sudo mkdir -p /etc/pki/tls/certs
sudo ln -s /etc/ssl/certs/ca-certificates.crt /etc/pki/tls/certs/ca-bundle.crt
```
(reference: http://stackoverflow.com/a/30154802/6358973)

I then installed from this tarball following the steps in my first post, but it seems that the build step `ant -f linux/build.xml` is done again with `debuild -uc -us`, is it?

**To summarize**:
```
cd where_you_want_to_install_lightzone
sudo apt-get install curl
curl -L https://github.com/Aries85/LightZone/tarball/master > lightzone_4.1.7.orig.tar.gz

# if curl returns an error:
# sudo mkdir -p /etc/pki/tls/certs
# sudo ln -s /etc/ssl/certs/ca-certificates.crt /etc/pki/tls/certs/ca-bundle.crt

tar zxvf lightzone_4.1.7.orig.tar.gz -C lightzone
cd lightzone

# I did this step but may be a duplicate of debuild -uc -us
# ant -f linux/build.xml
# launch a test run of lightzone:
# ant -f linux/build.xml run 

debuild -uc -us
sudo dpkg -i ../lightzone_4.1.7-0obs1_amd64.deb # the .deb package may have a different name in future releases
```

**Comment by ktgw0316** [comment]
```
...
/usr/bin/lightzone: 65: /usr/bin/lightzone: -XX:MinHeapFreeRatio=30: not found
Error: Could not find or load main class com.lightcrafts.platform.linux.LinuxLauncher
```
Sorry, that's my mistake. I've just fix it in faf6a80.

> it seems that the build step `ant -f linux/build.xml` is done again with `debuild -uc -us`, is it?

Yes, it is. So, you don't need to run `ant` before `debuild`.

Thank you for letting me know the info about CAFile. I'll add it to build instruction, too.

**Comment by calocedrus** [comment]
@ktgw0316 : perfect! Thank you.
A small detail though is that the directory tree in lightzone_4.1.7.orig.tar.gz is slightly modified compared to the previous tarball. So the latest tarball I've used expends now in an Aries85-LightZone-faf6a80 directory rather than directly into the directory of one's choice.
That is:
```
$ curl -L https://github.com/Aries85/LightZone/tarball/master > lightzone_4.1.7.orig.tar.gz
$ tar zxvf lightzone_4.1.7.orig.tar.gz -C lightzone
$ cd lightzone
$ ls
Aries85-LightZone-faf6a80
```
So:
```
debuild -uc -us
sudo dpkg -i ../lightzone_4.1.7-0obs1_amd64.deb
```
should be done in `Aries85-LightZone-faf6a80` and the tarball should be copied /moved in the parent directory of Aries85-LightZone-faf6a80.
Lightzone now launches without any problem.

**Comment by ktgw0316** [comment]
I updated the build instruction today, so I'm going to close this issue.

BTW,
> the tarball should be copied /moved in the parent directory

Alternatively, you can use `--strip-components=1` option:
```
tar xf lightzone_4.1.7.orig.tar.gz -C lightzone --strip-components=1
```
